### ISSUE TITLE
Missing pre-flattening blocks without unflattened IDs (Dead bush missing texture)

### ISSUE THREAD

**Issue by iceiix** [iceiix/stevenarella#89]
<img width="979" alt="screen shot 2019-01-12 at 8 35 31 am" src="https://user-images.githubusercontent.com/43691553/51075804-7dc30c80-1645-11e9-89ea-51bcb6f648a0.png">

<img width="966" alt="screen shot 2019-01-12 at 8 35 27 am" src="https://user-images.githubusercontent.com/43691553/51075807-80bdfd00-1645-11e9-93f9-076498cf055f.png">

Dead bush, as found in deserts, now has a missing texture. Probably caused by The Flattening #67 https://github.com/iceiix/steven/pull/67

**Comment by iceiix** [comment]
https://minecraft.gamepedia.com/Java_Edition_data_values/Pre-flattening#Block_IDs shows deadbush is 32 (pre-flattening, hierarchical). Logging block ids shows 31:0 - 33:14 are assigned:

1040 block state = Web hierarchical 30:0 offset=0
1041 block state = TallGrass { variant: DeadBush } hierarchical 31:0 offset=0
1042 block state = TallGrass { variant: TallGrass } hierarchical 31:1 offset=1
1043 block state = TallGrass { variant: Fern } hierarchical 31:2 offset=2
1057 block state = Piston { extended: false, facing: Up } hierarchical 33:1 offset=10

but no 32

```rust
    DeadBush {
        props {},
        offset None,
        material material::NON_SOLID,
        model { ("minecraft", "dead_bush") },
        collision vec![],
    }
```

offset None skips flattening, but it is also being skipped pre-flattening?

**Comment by iceiix** [comment]
get_hierarchical_data() returns `data` token, defaults to `Some(0)` if omitted
get_flat_offset() returns `offset` token, defaults to `Some(0)` if omitted

VanillaIDMap is initialized with `for block in iter {` in this conditional:

```rust
                        let offset = block.get_flat_offset();
                        if let Some(offset) = offset {
```

so it skips blocks not in the flattening, need to ensure these are still added in earlier versions

**Comment by iceiix** [comment]
Fixing vanilla_id registration if offset is none, also logging hierarchical blocks without flattened IDs, here's what this fix re-adds to pre-1.13.2:

```
+(no flat) block state = FlowingWater { level: 0 } hierarchical 8:0
+(no flat) block state = FlowingWater { level: 1 } hierarchical 8:1
+(no flat) block state = FlowingWater { level: 2 } hierarchical 8:2
+(no flat) block state = FlowingWater { level: 3 } hierarchical 8:3
+(no flat) block state = FlowingWater { level: 4 } hierarchical 8:4
+(no flat) block state = FlowingWater { level: 5 } hierarchical 8:5
+(no flat) block state = FlowingWater { level: 6 } hierarchical 8:6
+(no flat) block state = FlowingWater { level: 7 } hierarchical 8:7
+(no flat) block state = FlowingWater { level: 8 } hierarchical 8:8
+(no flat) block state = FlowingWater { level: 9 } hierarchical 8:9
+(no flat) block state = FlowingWater { level: 10 } hierarchical 8:10
+(no flat) block state = FlowingWater { level: 11 } hierarchical 8:11
+(no flat) block state = FlowingWater { level: 12 } hierarchical 8:12
+(no flat) block state = FlowingWater { level: 13 } hierarchical 8:13
+(no flat) block state = FlowingWater { level: 14 } hierarchical 8:14
+(no flat) block state = FlowingWater { level: 15 } hierarchical 8:15
+(no flat) block state = FlowingLava { level: 0 } hierarchical 10:0
+(no flat) block state = FlowingLava { level: 1 } hierarchical 10:1
+(no flat) block state = FlowingLava { level: 2 } hierarchical 10:2
+(no flat) block state = FlowingLava { level: 3 } hierarchical 10:3
+(no flat) block state = FlowingLava { level: 4 } hierarchical 10:4
+(no flat) block state = FlowingLava { level: 5 } hierarchical 10:5
+(no flat) block state = FlowingLava { level: 6 } hierarchical 10:6
+(no flat) block state = FlowingLava { level: 7 } hierarchical 10:7
+(no flat) block state = FlowingLava { level: 8 } hierarchical 10:8
+(no flat) block state = FlowingLava { level: 9 } hierarchical 10:9
+(no flat) block state = FlowingLava { level: 10 } hierarchical 10:10
+(no flat) block state = FlowingLava { level: 11 } hierarchical 10:11
+(no flat) block state = FlowingLava { level: 12 } hierarchical 10:12
+(no flat) block state = FlowingLava { level: 13 } hierarchical 10:13
+(no flat) block state = FlowingLava { level: 14 } hierarchical 10:14
+(no flat) block state = FlowingLava { level: 15 } hierarchical 10:15
+(no flat) block state = Log { variant: Oak, axis: None } hierarchical 17:12
+(no flat) block state = Log { variant: Spruce, axis: None } hierarchical 17:13
+(no flat) block state = Log { variant: Birch, axis: None } hierarchical 17:14
+(no flat) block state = Log { variant: Jungle, axis: None } hierarchical 17:15
+(no flat) block state = Leaves { variant: Oak, decayable: false, check_decay: true, distance: 1 } hierarchical 18:8
+(no flat) block state = Leaves { variant: Spruce, decayable: false, check_decay: true, distance: 1 } hierarchical 18:9
+(no flat) block state = Leaves { variant: Birch, decayable: false, check_decay: true, distance: 1 } hierarchical 18:10
+(no flat) block state = Leaves { variant: Jungle, decayable: false, check_decay: true, distance: 1 } hierarchical 18:11
+(no flat) block state = Leaves { variant: Oak, decayable: true, check_decay: true, distance: 1 } hierarchical 18:12
+(no flat) block state = Leaves { variant: Spruce, decayable: true, check_decay: true, distance: 1 } hierarchical 18:13
+(no flat) block state = Leaves { variant: Birch, decayable: true, check_decay: true, distance: 1 } hierarchical 18:14
+(no flat) block state = Leaves { variant: Jungle, decayable: true, check_decay: true, distance: 1 } hierarchical 18:15
+(no flat) block state = DeadBush hierarchical 32:0
+(no flat) block state = DoubleStoneSlab { seamless: false, variant: Stone } hierarchical 43:0
+(no flat) block state = DoubleStoneSlab { seamless: true, variant: Stone } hierarchical 43:8
+(no flat) block state = DoubleStoneSlab { seamless: false, variant: Sandstone } hierarchical 43:1
+(no flat) block state = DoubleStoneSlab { seamless: true, variant: Sandstone } hierarchical 43:9
+(no flat) block state = DoubleStoneSlab { seamless: false, variant: PetrifiedWood } hierarchical 43:2
+(no flat) block state = DoubleStoneSlab { seamless: false, variant: Cobblestone } hierarchical 43:3
+(no flat) block state = DoubleStoneSlab { seamless: false, variant: Brick } hierarchical 43:4
+(no flat) block state = DoubleStoneSlab { seamless: false, variant: StoneBrick } hierarchical 43:5
+(no flat) block state = DoubleStoneSlab { seamless: false, variant: NetherBrick } hierarchical 43:6
+(no flat) block state = DoubleStoneSlab { seamless: false, variant: Quartz } hierarchical 43:7
+(no flat) block state = DoubleStoneSlab { seamless: true, variant: Quartz } hierarchical 43:15
+(no flat) block state = StoneSlab { half: Top, variant: Stone } hierarchical 44:8
+(no flat) block state = StoneSlab { half: Bottom, variant: Stone } hierarchical 44:0
+(no flat) block state = StoneSlab { half: Top, variant: Sandstone } hierarchical 44:9
+(no flat) block state = StoneSlab { half: Bottom, variant: Sandstone } hierarchical 44:1
+(no flat) block state = StoneSlab { half: Top, variant: PetrifiedWood } hierarchical 44:10
+(no flat) block state = StoneSlab { half: Bottom, variant: PetrifiedWood } hierarchical 44:2
+(no flat) block state = StoneSlab { half: Top, variant: Cobblestone } hierarchical 44:11
+(no flat) block state = StoneSlab { half: Bottom, variant: Cobblestone } hierarchical 44:3
+(no flat) block state = StoneSlab { half: Top, variant: Brick } hierarchical 44:12
+(no flat) block state = StoneSlab { half: Bottom, variant: Brick } hierarchical 44:4
+(no flat) block state = StoneSlab { half: Top, variant: StoneBrick } hierarchical 44:13
+(no flat) block state = StoneSlab { half: Bottom, variant: StoneBrick } hierarchical 44:5
+(no flat) block state = StoneSlab { half: Top, variant: NetherBrick } hierarchical 44:14
+(no flat) block state = StoneSlab { half: Bottom, variant: NetherBrick } hierarchical 44:6
+(no flat) block state = StoneSlab { half: Top, variant: Quartz } hierarchical 44:15
+(no flat) block state = StoneSlab { half: Bottom, variant: Quartz } hierarchical 44:7
+(no flat) block state = FurnaceLit { facing: North } hierarchical 62:2
+(no flat) block state = FurnaceLit { facing: South } hierarchical 62:3
+(no flat) block state = FurnaceLit { facing: West } hierarchical 62:4
+(no flat) block state = FurnaceLit { facing: East } hierarchical 62:5
+(no flat) block state = RedstoneOreLit hierarchical 74:0
+(no flat) block state = RedstoneTorchUnlit { facing: East } hierarchical 75:1
+(no flat) block state = RedstoneTorchUnlit { facing: West } hierarchical 75:2
+(no flat) block state = RedstoneTorchUnlit { facing: South } hierarchical 75:3
+(no flat) block state = RedstoneTorchUnlit { facing: North } hierarchical 75:4
+(no flat) block state = RedstoneTorchUnlit { facing: Up } hierarchical 75:5
+(no flat) block state = RedstoneTorchLit { facing: East } hierarchical 76:1
+(no flat) block state = RedstoneTorchLit { facing: West } hierarchical 76:2
+(no flat) block state = RedstoneTorchLit { facing: South } hierarchical 76:3
+(no flat) block state = RedstoneTorchLit { facing: North } hierarchical 76:4
+(no flat) block state = RedstoneTorchLit { facing: Up } hierarchical 76:5
+(no flat) block state = PumpkinFace { facing: North, without_face: false } hierarchical 86:2
+(no flat) block state = PumpkinFace { facing: South, without_face: false } hierarchical 86:0
+(no flat) block state = PumpkinFace { facing: West, without_face: false } hierarchical 86:1
+(no flat) block state = PumpkinFace { facing: East, without_face: false } hierarchical 86:3
+(no flat) block state = PumpkinFace { facing: North, without_face: true } hierarchical 86:6
+(no flat) block state = PumpkinFace { facing: South, without_face: true } hierarchical 86:4
+(no flat) block state = PumpkinFace { facing: West, without_face: true } hierarchical 86:5
+(no flat) block state = PumpkinFace { facing: East, without_face: true } hierarchical 86:7
+(no flat) block state = PumpkinLit { facing: North, without_face: true } hierarchical 91:6
+(no flat) block state = PumpkinLit { facing: South, without_face: true } hierarchical 91:4
+(no flat) block state = PumpkinLit { facing: West, without_face: true } hierarchical 91:5
+(no flat) block state = PumpkinLit { facing: East, without_face: true } hierarchical 91:7
+(no flat) block state = RepeaterPowered { delay: 1, facing: North, locked: false } hierarchical 94:2
+(no flat) block state = RepeaterPowered { delay: 2, facing: North, locked: false } hierarchical 94:6
+(no flat) block state = RepeaterPowered { delay: 3, facing: North, locked: false } hierarchical 94:10
+(no flat) block state = RepeaterPowered { delay: 4, facing: North, locked: false } hierarchical 94:14
+(no flat) block state = RepeaterPowered { delay: 1, facing: South, locked: false } hierarchical 94:0
+(no flat) block state = RepeaterPowered { delay: 2, facing: South, locked: false } hierarchical 94:4
+(no flat) block state = RepeaterPowered { delay: 3, facing: South, locked: false } hierarchical 94:8
+(no flat) block state = RepeaterPowered { delay: 4, facing: South, locked: false } hierarchical 94:12
+(no flat) block state = RepeaterPowered { delay: 1, facing: West, locked: false } hierarchical 94:1
+(no flat) block state = RepeaterPowered { delay: 2, facing: West, locked: false } hierarchical 94:5
+(no flat) block state = RepeaterPowered { delay: 3, facing: West, locked: false } hierarchical 94:9
+(no flat) block state = RepeaterPowered { delay: 4, facing: West, locked: false } hierarchical 94:13
+(no flat) block state = RepeaterPowered { delay: 1, facing: East, locked: false } hierarchical 94:3
+(no flat) block state = RepeaterPowered { delay: 2, facing: East, locked: false } hierarchical 94:7
+(no flat) block state = RepeaterPowered { delay: 3, facing: East, locked: false } hierarchical 94:11
+(no flat) block state = RepeaterPowered { delay: 4, facing: East, locked: false } hierarchical 94:15
+(no flat) block state = BrownMushroomBlock { is_stem: true, west: false, up: false, south: false, north: false, east: false, down: false } hierarchical 99:15
+(no flat) block state = RedMushroomBlock { is_stem: true, west: false, up: false, south: false, north: false, east: false, down: false } hierarchical 100:15
+(no flat) block state = DoubleWoodenSlab { variant: Oak } hierarchical 125:0
+(no flat) block state = DoubleWoodenSlab { variant: Spruce } hierarchical 125:1
+(no flat) block state = DoubleWoodenSlab { variant: Birch } hierarchical 125:2
+(no flat) block state = DoubleWoodenSlab { variant: Jungle } hierarchical 125:3
+(no flat) block state = DoubleWoodenSlab { variant: Acacia } hierarchical 125:4
+(no flat) block state = DoubleWoodenSlab { variant: DarkOak } hierarchical 125:5
+(no flat) block state = WoodenSlab { half: Top, variant: Oak } hierarchical 126:8
+(no flat) block state = WoodenSlab { half: Bottom, variant: Oak } hierarchical 126:0
+(no flat) block state = WoodenSlab { half: Top, variant: Spruce } hierarchical 126:9
+(no flat) block state = WoodenSlab { half: Bottom, variant: Spruce } hierarchical 126:1
+(no flat) block state = WoodenSlab { half: Top, variant: Birch } hierarchical 126:10
+(no flat) block state = WoodenSlab { half: Bottom, variant: Birch } hierarchical 126:2
+(no flat) block state = WoodenSlab { half: Top, variant: Jungle } hierarchical 126:11
+(no flat) block state = WoodenSlab { half: Bottom, variant: Jungle } hierarchical 126:3
+(no flat) block state = WoodenSlab { half: Top, variant: Acacia } hierarchical 126:12
+(no flat) block state = WoodenSlab { half: Bottom, variant: Acacia } hierarchical 126:4
+(no flat) block state = WoodenSlab { half: Top, variant: DarkOak } hierarchical 126:13
+(no flat) block state = WoodenSlab { half: Bottom, variant: DarkOak } hierarchical 126:5
+(no flat) block state = Tripwire { powered: false, attached: false, disarmed: false, north: false, south: false, west: false, east: false, mojang_cant_even: true } hierarchical 132:2
+(no flat) block state = Tripwire { powered: true, attached: false, disarmed: false, north: false, south: false, west: false, east: false, mojang_cant_even: true } hierarchical 132:3
+(no flat) block state = Tripwire { powered: false, attached: true, disarmed: false, north: false, south: false, west: false, east: false, mojang_cant_even: true } hierarchical 132:6
+(no flat) block state = Tripwire { powered: true, attached: true, disarmed: false, north: false, south: false, west: false, east: false, mojang_cant_even: true } hierarchical 132:7
+(no flat) block state = Tripwire { powered: false, attached: false, disarmed: true, north: false, south: false, west: false, east: false, mojang_cant_even: true } hierarchical 132:10
+(no flat) block state = Tripwire { powered: true, attached: false, disarmed: true, north: false, south: false, west: false, east: false, mojang_cant_even: true } hierarchical 132:11
+(no flat) block state = Tripwire { powered: false, attached: true, disarmed: true, north: false, south: false, west: false, east: false, mojang_cant_even: true } hierarchical 132:14
+(no flat) block state = Tripwire { powered: true, attached: true, disarmed: true, north: false, south: false, west: false, east: false, mojang_cant_even: true } hierarchical 132:15
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 1 } hierarchical 140:1
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 2 } hierarchical 140:2
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 3 } hierarchical 140:3
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 4 } hierarchical 140:4
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 5 } hierarchical 140:5
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 6 } hierarchical 140:6
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 7 } hierarchical 140:7
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 8 } hierarchical 140:8
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 9 } hierarchical 140:9
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 10 } hierarchical 140:10
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 11 } hierarchical 140:11
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 12 } hierarchical 140:12
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 13 } hierarchical 140:13
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 14 } hierarchical 140:14
+(no flat) block state = FlowerPot { contents: Empty, legacy_data: 15 } hierarchical 140:15
+(no flat) block state = SkullSkeletonWall { facing: Up, nodrop: false } hierarchical 144:1
+(no flat) block state = ComparatorPowered { facing: North, mode: Compare, powered: false } hierarchical 150:2
+(no flat) block state = ComparatorPowered { facing: South, mode: Compare, powered: false } hierarchical 150:0
+(no flat) block state = ComparatorPowered { facing: West, mode: Compare, powered: false } hierarchical 150:1
+(no flat) block state = ComparatorPowered { facing: East, mode: Compare, powered: false } hierarchical 150:3
+(no flat) block state = ComparatorPowered { facing: North, mode: Subtract, powered: false } hierarchical 150:6
+(no flat) block state = ComparatorPowered { facing: South, mode: Subtract, powered: false } hierarchical 150:4
+(no flat) block state = ComparatorPowered { facing: West, mode: Subtract, powered: false } hierarchical 150:5
+(no flat) block state = ComparatorPowered { facing: East, mode: Subtract, powered: false } hierarchical 150:7
+(no flat) block state = ComparatorPowered { facing: North, mode: Compare, powered: true } hierarchical 150:10
+(no flat) block state = ComparatorPowered { facing: South, mode: Compare, powered: true } hierarchical 150:8
+(no flat) block state = ComparatorPowered { facing: West, mode: Compare, powered: true } hierarchical 150:9
+(no flat) block state = ComparatorPowered { facing: East, mode: Compare, powered: true } hierarchical 150:11
+(no flat) block state = ComparatorPowered { facing: North, mode: Subtract, powered: true } hierarchical 150:14
+(no flat) block state = ComparatorPowered { facing: South, mode: Subtract, powered: true } hierarchical 150:12
+(no flat) block state = ComparatorPowered { facing: West, mode: Subtract, powered: true } hierarchical 150:13
+(no flat) block state = ComparatorPowered { facing: East, mode: Subtract, powered: true } hierarchical 150:15
+(no flat) block state = Leaves2 { check_decay: false, decayable: false, variant: Acacia } hierarchical 161:0
+(no flat) block state = Leaves2 { check_decay: true, decayable: false, variant: Acacia } hierarchical 161:8
+(no flat) block state = Leaves2 { check_decay: false, decayable: true, variant: Acacia } hierarchical 161:4
+(no flat) block state = Leaves2 { check_decay: true, decayable: true, variant: Acacia } hierarchical 161:12
+(no flat) block state = Leaves2 { check_decay: false, decayable: false, variant: DarkOak } hierarchical 161:1
+(no flat) block state = Leaves2 { check_decay: true, decayable: false, variant: DarkOak } hierarchical 161:9
+(no flat) block state = Leaves2 { check_decay: false, decayable: true, variant: DarkOak } hierarchical 161:5
+(no flat) block state = Leaves2 { check_decay: true, decayable: true, variant: DarkOak } hierarchical 161:13
+(no flat) block state = Log2 { axis: None, variant: Acacia } hierarchical 162:12
+(no flat) block state = Log2 { axis: X, variant: Acacia } hierarchical 162:4
+(no flat) block state = Log2 { axis: Y, variant: Acacia } hierarchical 162:0
+(no flat) block state = Log2 { axis: Z, variant: Acacia } hierarchical 162:8
+(no flat) block state = Log2 { axis: None, variant: DarkOak } hierarchical 162:13
+(no flat) block state = Log2 { axis: X, variant: DarkOak } hierarchical 162:5
+(no flat) block state = Log2 { axis: Y, variant: DarkOak } hierarchical 162:1
+(no flat) block state = Log2 { axis: Z, variant: DarkOak } hierarchical 162:9
+(no flat) block state = DaylightDetectorInverted { power: 0 } hierarchical 178:0
+(no flat) block state = DaylightDetectorInverted { power: 1 } hierarchical 178:1
+(no flat) block state = DaylightDetectorInverted { power: 2 } hierarchical 178:2
+(no flat) block state = DaylightDetectorInverted { power: 3 } hierarchical 178:3
+(no flat) block state = DaylightDetectorInverted { power: 4 } hierarchical 178:4
+(no flat) block state = DaylightDetectorInverted { power: 5 } hierarchical 178:5
+(no flat) block state = DaylightDetectorInverted { power: 6 } hierarchical 178:6
+(no flat) block state = DaylightDetectorInverted { power: 7 } hierarchical 178:7
+(no flat) block state = DaylightDetectorInverted { power: 8 } hierarchical 178:8
+(no flat) block state = DaylightDetectorInverted { power: 9 } hierarchical 178:9
+(no flat) block state = DaylightDetectorInverted { power: 10 } hierarchical 178:10
+(no flat) block state = DaylightDetectorInverted { power: 11 } hierarchical 178:11
+(no flat) block state = DaylightDetectorInverted { power: 12 } hierarchical 178:12
+(no flat) block state = DaylightDetectorInverted { power: 13 } hierarchical 178:13
+(no flat) block state = DaylightDetectorInverted { power: 14 } hierarchical 178:14
+(no flat) block state = DaylightDetectorInverted { power: 15 } hierarchical 178:15
+(no flat) block state = DoubleStoneSlab2 { seamless: false, variant: RedSandstone } hierarchical 181:0
+(no flat) block state = DoubleStoneSlab2 { seamless: true, variant: RedSandstone } hierarchical 181:8
+(no flat) block state = StoneSlab2 { half: Top, variant: RedSandstone } hierarchical 182:8
+(no flat) block state = StoneSlab2 { half: Bottom, variant: RedSandstone } hierarchical 182:0
+(no flat) block state = PurpurDoubleSlab { variant: Purpur } hierarchical 204:0
+(no flat) block state = PurpurSlab { half: Top, variant: Purpur } hierarchical 205:8
+(no flat) block state = PurpurSlab { half: Bottom, variant: Purpur } hierarchical 205:0
+(no flat) block state = Missing253 hierarchical 253:0
+(no flat) block state = Missing254 hierarchical 254:0
```

**Comment by iceiix** [comment]
<img width="966" alt="screen shot 2019-01-12 at 10 46 59 am" src="https://user-images.githubusercontent.com/43691553/51077173-7d337180-1657-11e9-8c86-05872fa5a004.png">
